### ISSUE TITLE
Using RMSEWithUncertainty in eval_metrics crashes a program

### ISSUE THREAD

**Issue by tsbalzhanov** [catboost/catboost#1698]
Problem:
catboost version: 0.25.1
Operating System: Ubuntu 18.04.5/macOS 10.15.7

Example code in Python
```python
from catboost import CatBoost, Pool

train_data = [["France", 1924, 44],
              ["USA", 1932, 37],
              ["USA", 1980, 37]]

data = [["USA", 1996, 197],
        ["France", 1968, 37],
        ["USA", 2002, 77]]

cat_features = [0]
train_label = [1, 1, 0]
label = [0, 0, 1]

train_dataset = Pool(data=train_data,
                     label=train_label,
                     cat_features=cat_features)
dataset = Pool(data=data,
               cat_features=cat_features,
               label=label)

model = CatBoost({'iterations': 10, 'loss_function': 'RMSE'})
model.fit(train_dataset, eval_set=dataset)
model.eval_metrics(dataset, ['RMSEWithUncertainty'])
```

**Comment by ek-ak** [comment]
Hello! Thanks for you issue! RMSEWithUncertainty couldn't be used with simple regression model (it needs 2-dimensional approx). We added check and it will fail without crash. In practice RMSEWithUncertainty metric should be used only with models fitted with RMSEWithUncertainty loss.

**Comment by andrey-khropov** [comment]
Fixed in release 0.26
